frontend/src/metabase/dashboard/actions/trash.ts

```diff
@@ -0,0 +1,101 @@
+import { t } from "ttag";
+import _ from "underscore";
+
+import { canonicalCollectionId } from "metabase/collections/utils";
+import Dashboards from "metabase/entities/dashboards";
+import { createThunkAction } from "metabase/lib/redux";
+import { addUndo } from "metabase/redux/undo";
+
+import { fetchDashboard } from "./data-fetching";
+
+// just using the entity action doesn't cause the dashboard to live update
+// calling fetchDashboard ensures that the view updates with the last values
+export const SET_ARCHIVED_DASHBOARD =
+  "metabase/dashboard/SET_ARCHIVED_DASHBOARD";
+export const setArchivedDashboard = createThunkAction(
+  SET_ARCHIVED_DASHBOARD,
+  function (archived = true) {
+    return async function (dispatch, getState) {
+      const { dashboardId } = getState().dashboard;
+
+      await dispatch(
+        Dashboards.actions.update({ id: dashboardId }, { archived }),
+      );
+
+      dispatch(
+        addUndo({
+          subject: t`dashboard`,
+          verb: archived ? t`trashed` : t`restored`,
```

Comment:
Word-for-word translations, oh dear. I'll leave it up to you whether this can be remedied

Reply:
Well, we had a requirements change, so I was able to fix this
```ts
export function getTrashUndoMessage(name: string, archived: boolean): string {
  return archived
    ? c(
        "{0} is the name of the entity being trashed, e.g. 'My Awesome Dashboard'",
      ).t`${name} has been moved to the trash.`
    : c(
        "{0} is the name of the entity being restored from the trashed, e.g. 'My Awesome Dashboard'",
      ).t`${name} has been restored.`;
}
```

Reply:
re-read dat guy: "restored from the trashed"

Reply:
btw the single quotes in the argument to `c` might trip up the verify-i18n step of CI